index.js

```diff
@@ -0,0 +1,91 @@
+var analyze = require('./lib/analyze-raptor');
+var Promise = require('promise');
+var fs = require('fs');
+
+var metric = 'visuallyLoaded';
+
+var readFile = function(dataFile, appContext, appEntryPoint) {
+  console.log('Processing %s', dataFile);
+  console.log('App (context: %s, entrypoint: %s)', appContext, appEntryPoint);
+  return new Promise(function(resolve, reject) {
+    fs.readFile(dataFile, { encoding: 'utf8' }, function(err, content) {
+      if (err) {
+        return reject(err);
+      }
+      var data = content.split('\n');
+      data = JSON.parse(data);
+      resolve(data);
+    });
+  });
+};
+
+var parseAppData = function(dataObj, appContext, appEntryPoint) {
+  return new Promise(function(resolve, reject) {
+    var perfData = [];
+    var series = dataObj.results[0].series;
+
+    series.forEach(function(entry){
+      var current = entry;
+      // only want measures 
+      if (current.name !== 'measure') {
+        return;
+      }
+
+      // step thru each measure entry
+      current.values.forEach(function(measure){
+        // only want specific app & entrypoint (if provided)
+        if (appEntryPoint !== 'none') {
+          var fullName = appContext + '@' + appEntryPoint;
+          if (measure[2] !== fullName) {
+            return;
+          }
+        } else {
+          if (measure[2] !== appContext) {
+            return;
+          }
+        }
+        // only interested in specified metric ie. visuallyLoaded
+        if (measure[6] !== metric) {
+          return;
+        }
+        perfData.push({
+          time: measure[4],
+          value: measure[10],
+          revisionId: measure[8]
+        });
+      });
+
+    });
+    resolve(perfData);
+  });
+};
+
+var raptorRegression = function(dataFile, appContext, appEntryPoint) {
+  return new Promise(function(resolve, reject) {
+    var regressionList = [];
+    readFile(dataFile, appContext, appEntryPoint)
+      .then(function(dataObj) {
+        return parseAppData(dataObj, appContext, appEntryPoint);
+      })
+      .then(function(perfData) {
+        if (!perfData.length) {
+          throw new Error("No '%s' meaures found for app (context: %s, entrypoint: %s)");
+        }
+        return analyze(perfData);
+      })
+      .then(function(regressionList) {
+        if (regressionList.length == 0) {
+          console.log("No regressions detected.");
+        } else {
+          console.log("Regressions detected.");
+        }
+        resolve(regressionList);
+      })
+      .catch(function(err) {
+        console.error(err.stack || err);
```

Comment:
In order for these errors to "bubble up", we need to `reject` here:

```js
.catch(reject);
```

Then you can move this logic into `bin/raptor-regression` and `.catch()` there.